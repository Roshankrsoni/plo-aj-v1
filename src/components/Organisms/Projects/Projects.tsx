import React from "react";
import ProjectCard from "../../Molecules/ProjectCard/ProjectCard";
import "./Projects.css";
import { useGlobalContext } from "../../../context/GlobalContext";
import SectionVisibilityHOC from "../SectionVisibilityHOC/SectionVisibilityHOC";

function Projects() {
  const { profileData } = useGlobalContext();
  const { projects } = profileData;

  return (
    <>
      <div className="section-title h2 bold">Fun Learning Projects</div>
      <div className="subsection projectsectionbody">
        {projects.map((projectDetails, index) => {
          return <ProjectCard key={index} projectDetails={projectDetails} />;
        })}
      </div>
    </>
  );
}

export default SectionVisibilityHOC(Projects, "projects");
