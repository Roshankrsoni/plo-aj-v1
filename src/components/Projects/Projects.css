.projectsectionbody{
    display:flex;
    width: 100%;
    padding-bottom: 0.5rem;
    transition: all 0.5s ease-in-out;
}
.showall{
    flex-wrap: wrap;
}
.showall .projectcard{
    width: 100%;;
}
.showsome{
    flex-wrap: nowrap;
    overflow-y: scroll;
}

/* .projectsectionbody::-webkit-scrollbar {
    display: none;
} */

.projectcard + .projectcard{
    margin-left: 0.5rem;
}

.showall .projectcard + .projectcard{
    margin-top:0.5rem;
    margin-left: 0;
}

.projects .scroll-text{
    padding-top:1.5rem;
    cursor: pointer;
}


/* width */
.projectsectionbody::-webkit-scrollbar {
width: 0;
height:4px;
}

/* Track */
.projectsectionbody::-webkit-scrollbar-track {
background: white;
}

/* Handle */
.projectsectionbody::-webkit-scrollbar-thumb {
background: #f2f2f2;
border-radius: 10px;
}

/* Handle on hover */
.projectsectionbody::-webkit-scrollbar-thumb:hover {
background: #333333;
}


/* Track */
.darkmode .projectsectionbody::-webkit-scrollbar-track {
background: #333333;

}

/* Handle */
.darkmode  .projectsectionbody::-webkit-scrollbar-thumb {
background: #4F4F4F;
border-radius: 10px;
}

/* Handle on hover */
.darkmode  .projectsectionbody::-webkit-scrollbar-thumb:hover {
background: #333333;
}


  