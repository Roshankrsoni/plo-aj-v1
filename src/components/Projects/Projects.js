import React, { useState, useEffect }from "react";
import ProjectCard from "../ProjectCard/ProjectCard";

import "./Projects.css";

function Projects(props) {
    
    let [viewAll, setViewAll] = useState();
    useEffect(()=>{
        setViewAll(false);
    },[])
    
    function viewAllHandler(){
        if(viewAll){
            setViewAll(false);
        }else{
            setViewAll(true);
        }
       
    }

  return (
    <div
      className={
        "show-on-scroll col-md-7 page-1 text-left section projects " +
        props.class
      }
    >
      <div className="section-title grey4 h2 bold">Recent Projects</div>
      
      <div className={ `subsection projectsectionbody ${viewAll ? 'showall' : 'showsome'}`}>
          {
              props.projects && props.projects.map((ele,index) => {
                  return (
                      <ProjectCard key={index} data={ele}/>
                  )
              })
          }
      </div>
      <div className="default-text red scroll-text thin" onClick={viewAllHandler}>
          {
              viewAll ? 'Collapse Section' : 'View All Projects'
          }
        </div>

    </div>
  );
}

export default Projects;
