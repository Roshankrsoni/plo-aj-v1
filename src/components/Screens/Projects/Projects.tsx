import React from "react";
import ProjectCard from "../../Molecules/ProjectCard/ProjectCard";
import "./Projects.scss";
import { useGlobalContext } from "../../../context/GlobalContext";
import SectionVisibilityHOC from "../../Organisms/SectionVisibilityHOC/SectionVisibilityHOC";
import T from "../../../translations/en_IN";
import constants from "../../../helpers/constants";
function Projects() {
  const { profileData } = useGlobalContext();
  const { projects } = profileData;

  return (
    <>
      <div className="section-title h2 bold">{T.FUN_LEARNING_PROJECTS}</div>
      <div className="subsection project-section-body">
        {projects.map((projectDetails, index) => {
          return (
            <ProjectCard key={String(index)} projectDetails={projectDetails} />
          );
        })}
      </div>
    </>
  );
}

export default SectionVisibilityHOC(Projects, constants.classNames.PROJECTS);
