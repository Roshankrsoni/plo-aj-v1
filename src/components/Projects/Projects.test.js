import React from 'react';
import { shallow } from 'enzyme';
import Projects from './Projects';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Projects dbData={dummyData} />);
        expect(component).toMatchSnapshot();
    });
});