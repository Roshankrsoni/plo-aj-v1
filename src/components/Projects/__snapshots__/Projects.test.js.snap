// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="show-on-scroll col-md-7 page-1 text-left section projects undefined"
>
  <div
    className="section-title grey4 h2 bold"
  >
    Recent Projects
  </div>
  <div
    className="subsection projectsectionbody showsome"
  />
  <div
    className="default-text red scroll-text thin"
    onClick={[Function]}
  >
    View All Projects
  </div>
</div>
`;
